### ISSUE TITLE
unable to import from keepass file (.kdb)

### ISSUE THREAD

**Issue by hothouse-github** [tildaslash/RatticWeb#306]
Hi RatticDB Team,

Have tried importing one of our keepass databases (which is in a kbdx format).
First task was to export to a kdb format (as it doesn't seem to support kdbx).
Validated that the exported file was valid (using keepass 1.27).

However when importing into rattic the following error is thrown (via a 500 internal server error in apache2):

2014-09-12 12:58:00,002 [ERROR] Internal Server Error: /staff/import/keepass/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 199, in get_response
    response = middleware_method(request, response)
  File "/usr/local/lib/python2.7/dist-packages/user_sessions/middleware.py", line 46, in process_response
    request.session.save()
  File "/usr/local/lib/python2.7/dist-packages/user_sessions/backends/db.py", line 69, in save
    session_data=self.encode(self._get_session(no_load=must_create)),
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/sessions/backends/base.py", line 87, in encode
    serialized = self.serializer().dumps(session_dict)
  File "/usr/local/lib/python2.7/dist-packages/django/core/signing.py", line 88, in dumps
    return json.dumps(obj, separators=(',', ':')).encode('latin-1')
  File "/usr/lib/python2.7/json/__init__.py", line 238, in dumps
    *_kw).encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python2.7/json/encoder.py", line 263, in iterencode
    return _iterencode(o, 0)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xb4 in position 0: invalid start byte
[client 192.168.X.X] mod_wsgi (pid=1287): Exception occurred processing WSGI script '/home/ratticdb/application/RatticWeb/ratticweb/wsgi.py'.
[client 192.168.X.X] Traceback (most recent call last):
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/wsgi.py", line 206, in __call__
[client 192.168.X.X]     response = self.get_response(request)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 203, in get_response
[client 192.168.X.X]     response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 236, in handle_uncaught_exception
[client 192.168.X.X]     return callback(request, *_param_dict)
[client 192.168.X.X]   File "/home/ratticdb/application/RatticWeb/ratticweb/views.py", line 14, in handle500
[client 192.168.X.X]     return render(request, '500.html', status=500)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/**init**.py", line 53, in render
[client 192.168.X.X]     return HttpResponse(loader.render_to_string(_args, *_kwargs),
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 162, in render_to_string
[client 192.168.X.X]     t = get_template(template_name)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 138, in get_template
[client 192.168.X.X]     template, origin = find_template(template_name)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 127, in find_template
[client 192.168.X.X]     source, display_name = loader(name, dirs)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 43, in **call**
[client 192.168.X.X]     return self.load_template(template_name, template_dirs)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 49, in load_template
[client 192.168.X.X]     template = get_template_from_string(source, origin, template_name)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 149, in get_template_from_string
[client 192.168.X.X]     return Template(source, origin, name)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 125, in **init**
[client 192.168.X.X]     self.nodelist = compile_string(template_string, origin)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 153, in compile_string
[client 192.168.X.X]     return parser.parse()
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 278, in parse
[client 192.168.X.X]     compiled_result = compile_func(self, token)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 215, in do_extends
[client 192.168.X.X]     nodelist = parser.parse()
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 278, in parse
[client 192.168.X.X]     compiled_result = compile_func(self, token)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 190, in do_block
[client 192.168.X.X]     nodelist = parser.parse(('endblock',))
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 276, in parse
[client 192.168.X.X]     self.invalid_block_tag(token, command, parse_until)
[client 192.168.X.X]   File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 331, in invalid_block_tag
[client 192.168.X.X](command, get_text_list%28["'%s'" % p for p in parse_until]%29))
[client 192.168.X.X] TemplateSyntaxError: Invalid block tag: 'trans', expected 'endblock'

Thoughts ?

**Comment by enuno** [comment]
I am having the same issue. Is there a fix for this?

**Comment by ghost** [comment]
@smarthall I've just run into this problem and found out that Rattic uses session to pass file data as a python object(?). The problem is Django >1.6 uses JSONSerializer as the default and it doesn't support that kind of complex types. So, by forcing ratticweb app's SESSION_SERIALIZER to PickleSerializer, I've fixed my problem.

**tl;dr**
Add this line to ratticweb/settings.py and everything will be fine:
`SESSION_SERIALIZER = 'django.contrib.sessions.serializers.PickleSerializer'`

**Comment by iddqd2** [comment]
Using v1.3.1.
It's already have that line in settings.py.
Import with cyrrilic still fails with the following error: 
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-12: ordinal not in range(128)

**Comment by jasonmarshall1** [comment]
I encountered the 500 server error and was able to import the keepass 2.34 file into Rattic by following the following process:  
1) create keepass files one for each folder under the root directory in keepass. When I tried to export/import a keepass file with subdirectories/folders setup in keepass, the import puked. 
2) export each file using V1 option for each directory/folder in keepass
3) this tripped me up, but use the finder feature to navigate to the export directory. When I typed in the folder and file name, it failed. I'm not sure what is different, but using the finder to locate the directory and then typing in the export file name seemed to work. My file extension was .kdb
4) I used the key file (yourKeePassFile.key) key value when prompted for the password during the import in rattic, not the user supplied pwd. 

I hope this helps someone else.
